ui/src/components/credentials/ObjectAttributeValueTreeNode.tsx

```diff
@@ -0,0 +1,90 @@
+import { Col, Row, Space, Typography } from "antd";
+
+import { z } from "zod";
+import { ReactComponent as IconTreeLeaf } from "src/assets/icons/file-04.svg";
+import { ReactComponent as IconTreeNode } from "src/assets/icons/folder.svg";
+import { AttributeValue, JsonLiteral } from "src/domain";
+import { formatDate } from "src/utils/forms";
+
+function extractValue(attributeValue: AttributeValue): JsonLiteral | undefined {
+  switch (attributeValue.type) {
+    case "integer":
+    case "number":
+    case "null":
+    case "boolean": {
+      return attributeValue.value === undefined ? "-" : attributeValue.value;
+    }
+    case "string": {
+      const value = attributeValue.value;
+      if (value === undefined) {
+        return "-";
+      } else {
+        switch (attributeValue.schema.format) {
+          case "date":
+          case "date-time":
+          case "time": {
+            const parsedDate = z.coerce.date(z.string().datetime()).safeParse(value);
+            return parsedDate.success
+              ? formatDate(parsedDate.data, attributeValue.schema.format)
+              : attributeValue.value;
+          }
+          default: {
+            return attributeValue.value === undefined ? "-" : attributeValue.value;
+          }
+        }
+      }
+    }
+    case "object": {
+      return undefined;
```

Comment:
This shouldn't be `null`?

Reply:
No, because we have an attribute whose type is `null`. This `undefined` represents the absence of a value, and we need to differentiate it from the legit value `null` and the non-provided value for optional attributes which is represented by the string literal `"-"`.